### PR TITLE
feat(): Replacement for getKlass utility

### PR DESCRIPTION
**Pull request by asturur**
## Motivation

fabric.util.getKlass required all the objects to be imported in a fabric global variable in order to be restored from json.
This made impossible to split the code you are not using with imports.
This small system of registering class when imported allow for building apps that require just some parts of fabricJS without importing the whole bundle.

The feature is written just with this in mind and can be extended later with more nuances ( as @ShaMan123 custom resolvers for some classes #8282 )

### CODE REVIEW

**Comment by asturur** [comment]
Ok i have some issues with Pattern that i didn't account for ( not understand why not for Gradient then... ) i ll fix it asap.
The extra 300bytes of code for this are mostly for all the explict calls we do, i don't think i can avoid it. ( 40 calls to .setClass can't be really minified )

**Review by ShaMan123 — COMMENTED** [review]
I don't see a real difference between #8282 to this PR. 
Over there I had more freedom to solve things like Gradient not having a type and I did more work to adapt fabric not to import classes, but instead call the registry (such as in `enlivenObjectEnlivables` and in the `elements_parser`)

File: src/shapes/circle.class.ts [hunk 1]
```diff
@@ -220,4 +221,7 @@ export const circleDefaultValues: Partial<TClassProperties<Circle>> = {
 
 Object.assign(Circle.prototype, circleDefaultValues);
 
+classRegistry.setClass(Circle);
+classRegistry.setSVGClass(Circle);
```
[the remark this answers is not in the document]
Comment (ShaMan123) [reply src/shapes/circle.class.ts:225]: Or better, why not check if `fromElement` exists...
#8282

**Comment by asturur** [comment]
i remember #8282 was huge. Let me look again

File: src/shapes/circle.class.ts [hunk 2]
```diff
@@ -220,4 +221,7 @@ export const circleDefaultValues: Partial<TClassProperties<Circle>> = {
...
+
 fabric.Circle = Circle;
```
[the remark this answers is not in the document]
Comment (asturur) [reply src/shapes/circle.class.ts:227]: yes but now would make all tests fail, no?

**Comment by asturur** [comment]
Is hard to re-understand #8282 since now is showing diffs from an old situation, but i remember that the registry was resolving fromObject and fromElement instead that resolving the class and ther were changes to how we do fromObject and fromSVG to accomodate that. I just want to do only leaner/necessary changes now

File: src/shapes/circle.class.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShaMan123) [reply src/shapes/circle.class.ts:227]: not if we adjust the index file to use export, then rollup should build the fabric object by itself if I am not mistaken.

**Comment by ShaMan123** [comment]
#8282 had a large diff for the same reason this PR has, because each file must be registered. That is it.
Regarding what the registry returns, that is not a big deal IMO, though I think it makes it much more flexible.
But it is not the reason I pointed it out.
The registry itself is built in a way that svg tags or json types are easier to handle, meaning edge cases like Gradient is a one line fix.

**Comment by ShaMan123** [comment]
take a look at [`src/Registry.ts`](https://github.com/fabricjs/fabric.js/blob/5c5f26232da2be073ab500102b450c2a74a4f2d0/src/Registry.ts)
Someone could do registry.resolveJSONKey = (...args) => 'whatever'

I would remove the assert methods and throw in the getters.
My concept was to eliminate the need for a class to have `fromObject` `fromElement`, making things a lot more flexible. But that can wait. You can see that on top of the flexible methods I've exposed a `registerClass` method which is equivalent to calling `setClass` `setSVGClass` of this PR.
The diff is in the possibility of fine tuning the resolver.

**Pull request merged**